src/lib.rs

```diff
@@ -362,21 +362,28 @@ impl<W: Word> LinkerScript<W> {
 
     /// Generate a linker script and matching reset module
     /// which correctly initializes sections.
+    ///
+    /// The function places a linker script file, called `link.x`, in
+    /// the current working directory.
     pub fn generate(self) -> Result<()> {
-        const REQ_SEC_NAMES: [&str; 5] = ["stack", "vector_table", "text", "data", "bss"];
+        let mut link_x = File::create("link.x")?;
+        self.write(&mut link_x)
+    }
+
+    /// Write the linker script into the writer, `link_x`
+    pub fn write<Wr: Write>(self, link_x: &mut Wr) -> Result<()> {
+        const REQ_SEC_NAMES: [&str; 6] = ["stack", "vector_table", "text", "data", "rodata", "bss"];
```

Comment:
`rodata` is documented as a required section. I added it to this list, and checked it with some tests (below).